src/lib.rs

```diff
@@ -83,18 +96,46 @@ impl<'a, 'b> DrawingBackend for RatatuiBackend<'a, 'b> {
         coord1: BackendCoord,
         coord2: BackendCoord,
         style: &S,
-        _fill: bool,
+        fill: bool,
     ) -> std::result::Result<(), DrawingErrorKind<Self::ErrorType>> {
-        let (x1, y1) = backend_to_canvas_coords(coord1, self.size);
-        let (x2, y2) = backend_to_canvas_coords(coord2, self.size);
-
-        self.canvas.draw(&canvas::Rectangle {
-            x:      x1.min(x2),
-            y:      y1.min(y2),
-            width:  (x2 - x1).abs(),
-            height: (y2 - y1).abs(),
-            color:  convert_color(style.color()),
-        });
+        if fill {
+            let (mut start, mut stop) =
+                if coord1.0 < coord2.0 { (coord1, coord2) } else { (coord2, coord1) };
```

Comment:
Does this mean we don't care about the `coord1.0 < coord2.0 && coord1.1 > coord2.1`? Why aren't we ordering them independently?
```suggestion
            let (mut start, mut stop) = (
                (coord1.0.min(coord2.0), coord1.1.min(coord2.1)),
                (coord1.0.max(coord2.0), coord1.1.max(coord2.1)),
            );
```

Reply:
Independently ordering `coord1 = (2,3); coord2=(1,4);` would transformed them in to completely unrelated coordinates: `start = (1,3); stop = (2,4)`;

Reply:
Isn't it still the same rectangle? 
![image](https://github.com/SOF3/plotters-ratatui-backend/assets/19623715/eba8f7ca-5145-45cf-94b9-cff27b25351e)

Reply:
Yes, quite. I woke early today with my subconscious thoroughly embarrassed that yesterday wasn't a day for me to be doing even the simplest Cartesian geometry! 😳

Reply:
Done

Reply:
Hmmn.  This appears to work on a visual inspection of the tests I included, but not with my application.